toolchain/check/check.cpp

```diff
@@ -558,21 +558,21 @@ class DeferredDefinitionWorklist {
   llvm::SmallVector<Task, 0> worklist_;
 
   // A deferred definition scope that is currently still open.
-  struct EnclosingScope {
+  struct AncestorScope {
```

Comment:
I think it'd make sense to call this `OpenScope` or just `Scope`. Calling it "ancestor" makes me wonder what it's an ancestor of.

Reply:
How about `EnteredScope` / `entered_scopes_`? `Scope` seems a little too likely to conflict with a name elsewhere, "Open" doesn't quite feel right to me.

Reply:
Where I have LookupInAncestorScopes, would that be better talking about entered scopes too?

Reply:
`EnteredScope` looks good to me.

Hm. So, `LookupInAncestorScopes` performs name lookup in only the lexically-enclosing scopes that lack a `NameScopeId`, and also returns the list of ancestor scopes with a `NameScopeId` that might shadow the lexical lookup result. Maybe something like `LookupInLexicalScopes` might make sense? (`LookupInLexicalScopesAndFindNonLexicalScopes` would be more accurate but kinda long. But it only has one caller so maybe that's OK?)

Reply:
Going with LookupInLexicalScopes